### DISCUSSION TITLE
Shelly PLUS 1pm: Question about usage "SW" Input PIN

### DISCUSSION THREAD

**Discussion by Sysselix** [arendst/Tasmota#20622] [Q&A] [answered]
Hi all,

I 've just a Question about the usage "SW" Input PIN

I have successfully put Tasmota on my Shelly PLUS 1PM.
If I press on "Toggle" in the WebUI the relay toggles and put power on the Output PIN "O"

But the Shelly PLUS 1PM has another Input PIN called "SW"
I read here or there that the Shelly should toggle (Power ON) automatically "O" if on the SW are 230V
But it won't.

I have measured the voltage on the cords which are connected to L and N and my multimeter told me: 230V
I measured the voltage on the cords which are connected to N + SW and it's 230V too
The mains are connected permanent to the shelly with 230v

I have Tasmota Version 13.3.0(solo1)-2_0_14 (2023-12-12T14:31:38) installed on the Shelly PLUS 1PM and I ran auto-configuration.
This set the Template to "Shelly PLUS 1PM" like here: `https://templates.blakadder.com/shelly_plus_1PM.html`

I want to use the Shelly like here on this picture shown: If 230V on SW then power:on

[https://digitaldad.de/wp-content/uploads/2022/09/Shelly-Plus-1PM-Verkabelung-Schalter.png](https://digitaldad.de/wp-content/uploads/2022/09/Shelly-Plus-1PM-Verkabelung-Schalter.png)
Taken from: https://digitaldad.de/2022/10/03/shelly-plus-1pm-installation-konfiguration-einsatzmoeglichkeiten/

Maybe I forgot some config to do ?
Maybe someone could give me a hint ?

Thanks

-- 
Kind regards
Sysselix

**Comment by sfromis** [comment]
What happens depends on how you configure Tasmota. The standard template has the "SW" input configured as a Switch_n gpio.
https://templates.blakadder.com/shelly_plus_1PM.html
Default configuration for a Switch input is `SwitchMode 0` (toggle) which is inteded for a momentary push button input, toggling relay state every time you push/release the button.

If what you have is a locking toggle switch input, you should probably use `SwitchMode 1` (follow) instead.

**Reply by Sysselix** [reply]
I have already tried:
unpowering `SW` + entering `SwitchMode 1` on the console and repowering `SW`
But nothing happens.

I can't see on the console anything regarding the state of the input `SW`, even nothing with `weblog 4`
Maybe there is any out debug method I could use to see it `SW` is input powered or not ?
The cable itself has power (measured with my multimeter...

If I toggle the normal `Button` on the `WebUI` it will `Power ON` the output `O`
But I want to use the `SW Input` too...

I have exact the same setup like this (lying on the floor):
https://digitaldad.de/wp-content/uploads/2022/09/Shelly-Plus-1PM-Verkabelung-Schalter.png

![2024-01-30 15_18_33-Tasmota Cool Override - Configure Template – Mozilla Firefox](https://github.com/arendst/Tasmota/assets/136033764/af1bb753-cbde-41eb-b192-d604072e249c)

**Reply by sfromis** [reply]
Your gpio 25 config is bad, do not use same number twice for the same type. Most of the time, it makes more sense to have gpio 25 (the tiny button at the front corner of the case) configured as a Button gpio (to match the standard template), but you could also have it as a Switch, index 2.

I see that the autoconf setup does have this error, and have submitted a change. Seems that the templates site was updated later than the autoconf setup was created.

**Reply by Sysselix** [reply]
> Your gpio 25 config is bad

Ok, I tried already a little around, earlier before I opend the Thread here. But It doesn't resolved it.
I changed it to Button,1 now.

If I press the small button nothing happens (but the LED inside begins to blink)...

I have 2 of them (Shelly PLUS 1PM) and they both have the same misbehaviour.
They both can't be faulty...

And I still think that will hit all the Shelly PLUS (n)PM, because all of them (and maybe other devices too) have the same `SW Input Connector`

-- 
Kind regards
Sysselix

**Reply by sfromis** [reply]
I have a Shelly Plus 1PM working fine with the template config, switchmode 1 and a locking toggle between L and SW. Wiring matching the image you linked. The relay clicks (and lamp goes on/off) every time I toggle the switch. Similarly, each push on the on-device button toggles the light on/off.

If I add test rules like
`on Switch1#state do var1 %value% endon on Button1#state do var1 %value% endon`
I see the state changing to match the connection status of the toggle switch or button, instead of relay action.

Maybe you have been trying so many things that you have gotten to some combination not workable? Doing a `Reset 6` can be useful to reset everything (except for Wifi/MQTT) to default, you should then configure naming, template and SwitchMode after that.

Maybe you have a newer board revision not compatible with the one that the Tasmota setups were based on. Popping open the cover, mine has "20211026" and "Shelly 1PM+_v0.2.0". The methods mentioned here can be a way to "hunt" for unknown pin usage.
https://tasmota.github.io/docs/Configuration-Procedure-for-New-Devices/

But first, it could be useful to double-check if your switch physically is working, and with good connections on the wires.

**Comment by Sysselix** [comment]
> every time I toggle the switch. 

Which Switch ?
I just can Toggle via WebUI...

`Reset 6` done.
Autoconfig done
Set GPIO 25 to Button
`switchmode 1` set
But it won't turn on while `SW` has still 230V connected
It consumes no Power and the Switch in the WebUI is `OFF`

On the PCB ist printed: `20230428` and `Shelly 1PM+_v0.2.0`
(opening was so easy with my fingernails ^^)

I will try this tomorrow: Configuration procedure for new devices...

**Reply by sfromis** [reply]
The physical switch was what I meant. With same board version number, my expectation would not be that the board was revised, but can't say anything for certain.

Not sure what you mean by "it won't turn on while `SW` has still 230V connected", if you tried toggling it a few times. Does it block actions while 230V to the SW input? What about the testing rule, and using the gpio 25 button?

Strictly speaking, there is no "Switch" in the web UI, what you have there is the Power channel state for the relay output, which should react to Tasmota detecting Switch/Button action, when not configuring to not do that.

**Reply by Sysselix** [reply]
> it won't turn on while `SW` has still 230V connected

That just means: The cable is connected and 230v are still on the Input `SW` present/active.

I will continue testing on tomorrow...

Thanks so far and if someone has another Idea: Just tell me :-)

**Reply by Sysselix** [reply, marked as answer]
> I have a Shelly Plus 1PM working fine...

**You were right.**

Because of Your question about the board revision, I removed all cables and started over.
I added a hardware wall-light-switch and tried the small switch in the housing.
ALL WORKED FINE...

without setting any modes or rules.
It worked "out of the box" 👍 :-)

I'm a little confused now.
But now I have a working setup and can copy it to the other setup.

Thanks a lot for your great support !!!!!!!!!!
<party>

<closed>

**Reply by Sysselix** [reply]
Here some additional Infos about these both switches which are build in:
Maybe this will be work in Shelly PLUS 2PM / 3PM / 4PM etc...

`Switch_n - GPIO4` controls the SW Input PIN
`Switch (or better Button) on GPIO25` controles the Hardwarebutton which is mounted in the housing and controles the O (=Output)
You can set the GPIO25 as a button or as a switch, it workes both.
If GPIO25 is unset the hardware switch (mounted in the housing) will not work and turns not the Output an, either the Switch_n works fine.
